src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -903,6 +918,20 @@ public AppPage verifyHtmlMainContent(String filePath) throws IOException {
         return verifyHtmlPart(MAIN_CONTENT, filePath);
     }
 
+    public AppPage verifyHtmlMainContentWithReloadRetry(final String filePath) throws IOException {
+        return RetryManager.runUntilNoException(new RetryableTaskReturnsThrows<AppPage, IOException>("HTML verification") {
+            @Override
+            public AppPage run() throws IOException {
```

Comment:
The `Assertion` exception is more appropriate here. You can catch the IOException inside the method and rethrow assertion error.